app/src/main/java/de/westnordost/streetcomplete/util/LocaleContextWrapper.java

```diff
@@ -0,0 +1,75 @@
+package de.westnordost.streetcomplete.util;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.content.ContextWrapper;
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
+import android.os.Build;
+
+import androidx.preference.PreferenceManager;
+
+import java.util.Locale;
+
+import de.westnordost.streetcomplete.Prefs;
+
+/**
+ * @source https://stackoverflow.com/a/40704077
+ */
+public class LocaleContextWrapper extends ContextWrapper {
+
+	public LocaleContextWrapper(Context base) {
+		super(base);
+	}
+
+	@SuppressWarnings("deprecation")
+	public static ContextWrapper wrap(Context context) {
+		SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+
+		String language = prefs.getString(Prefs.LANGUAGE_SELECT, "");
+
+		Configuration config = context.getResources().getConfiguration();
+		Locale sysLocale = null;
+		if (Build.VERSION.SDK_INT > Build.VERSION_CODES.N) {
+			sysLocale = getSystemLocale(config);
+		} else {
+			sysLocale = getSystemLocaleLegacy(config);
+		}
+		if (!language.equals("") && !sysLocale.getLanguage().equals(language)) {
+			Locale locale = new Locale(language);
+			Locale.setDefault(locale);
+			if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+				setSystemLocale(config, locale);
+			} else {
+				setSystemLocaleLegacy(config, locale);
+			}
+
+		}
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+			context = context.createConfigurationContext(config);
+		} else {
+			context.getResources().updateConfiguration(config, context.getResources().getDisplayMetrics());
+		}
+		return new LocaleContextWrapper(context);
+	}
+
+	@SuppressWarnings("deprecation")
```

Comment:
Just to confirm: how serious this and other deprecations should be treated? Is there some existing replacement and deprecated things are used to support older phones?

Or is it deprecated without replacement and it is likely that in future the feature will no longer work?

Or is it deprecation that can be ignored?

Reply:
I have no clue :-) I am more a beginner Android Developer and it's source is stackoverflow... The good thing is that there is alternative code for newer version in there. Not sure if we could delete one version of the code, based on the build target.

Reply:
Looks like the legacy methods are only called on older api versions, where they will never be removed, so it should be safe.